apps/ewallet/lib/ewallet/trackers/transaction_tracker.ex

```diff
@@ -0,0 +1,109 @@
+# Copyright 2018-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EWallet.TransactionTracker do
+  @moduledoc """
+  This module is a GenServer started dynamically for a specific eWallet transaction
+  It will registers itself with the blockchain adapter to receive events about
+  a given transactions and act on it
+  """
+  use GenServer, restart: :temporary
+  require Logger
+
+  alias EWallet.BlockchainTransactionState
+  alias ActivityLogger.System
+
+  # TODO: handle failed transactions
+
+  def start_link(state) do
```

Comment:
The genserver should be building its own state rather than taking in external data directly. This makes it hard to keep track if things like `state[:registry]` or `state[:transaction]` is valid. But can take as a separate PR.